libevmasm/Assembly.cpp

```diff
@@ -407,7 +407,7 @@ map<u256, u256> Assembly::optimiseInternal(
 		if (_settings.runPeephole)
 		{
 			PeepholeOptimiser peepOpt(m_items);
-			while (peepOpt.optimise())
+			while (peepOpt.optimise() && count < 0xffffff)
```

Comment:
Do you have a test case which triggered a loop?

Reply:
No, but the two conditions (number of pops and bytecode size) could oscillate.

Reply:
Checked again, the won't as they currently are.